ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/SqlLedger.scala

```diff
@@ -396,6 +420,10 @@ private class SqlLedgerFactory(ledgerDao: LedgerDao) {
                 logger.warn(
                   s"Initial ledger entries provided, presumably from scenario, but I'm picking up from an existing database, and thus they will not be used")
               }
+              if (packages.listLfPackagesSync().nonEmpty) {
```

Comment:
I'd rather we don't thread the `InMemoryPackageStore` all the way to the Ledger initialization, but rather we simply initialize the ledger and automatically upload the packages from "the outside" via the regular mechanism. What do you think? Duplicate packages will be "ignored" and the other packages just uploaded.

Reply:
Not sure about that. This `initialize` method also inserts the provided ACS and ledger entries (generated by the scenario loader) into the database, and conceptually the packages should be added before the ledger entries.

In practice, nothing would happen if we first copied the given ledger entries and ACS upon initialization, and then uploaded the packages later from the outside.

Reply:
I see. Looking at my postgres-index(er) branch, I see that it won't be a problem. 👍